packages/slider/index.tsx

```diff
@@ -0,0 +1,380 @@
+          markerStyles[last] = clone;
+          return {markerStyles};
+        });
+      },
+      isRTL: (): boolean => this.props.isRtl || false,
+    };
+  }
+
+  static defaultProps: Partial<SliderProps> = {
+    className: '',
+    open: false,
+    min: 0,
+    max: 100,
+    value: 0,
+    step: 0,
+    discrete: false,
+    displayMarkers: false,
+    disabled: false,
+    tabIndex: 0,
+    isRtl: false,
+    onValueChange: () => {},
+  };
+  foundation!: MDCSliderFoundation;
+  sliderElement: React.RefObject<HTMLDivElement> = React.createRef();
+
+  state: SliderState = {
+    classList: new Set(),
+    markerStyles: [],
+    trackStyleProperty: {},
+    thumbContainerStyleProperty: {},
+  };
+
+  onDown?: (downEvent: MouseLikeEvent) => void;
+  onThumbDown?: () => void;
+  onThumbEnd?: () => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  onKeyDown?: (evt: React.KeyboardEvent) => void;
+
+  private transitionendEvtName = getCorrectEventName(
+    window,
+    'transitionend'
+  ) as EventType;
+
+  setStyleToElement = (
+    prop: string,
+    value: string | boolean,
+    elementStyleProperty: SliderElementNames
+  ) => {
+    const styleName = convertDashToCamelCase(prop);
+    const updateElementStyleProperty = Object.assign(
+      {},
+      this.state[elementStyleProperty],
+      {[styleName]: value}
+    );
+    this.setState((prevState) => {
+      return Object.assign(prevState, {
+        [elementStyleProperty]: updateElementStyleProperty,
+      });
+    });
+  };
+
+  componentDidMount() {
+    this.initFoundation();
+  }
+
+  componentDidUpdate(prevProps: SliderProps) {
+    const {step, value, min, max, disabled} = this.props;
+    if (disabled !== prevProps.disabled) {
+      this.foundation.setDisabled(disabled!);
+    }
+    if (step !== prevProps.step) {
+      this.foundation.setStep(step!);
+    }
+    if (value !== prevProps.value) {
+      this.foundation.setValue(value!);
+    }
+    if (min !== prevProps.min) {
+      this.foundation.setMin(min!);
+    }
+    if (max !== prevProps.max) {
+      this.foundation.setMax(max!);
+    }
+  }
+
+  componentWillUnmount() {
+    if (!this.foundation) {
+      return;
+    }
+    this.foundation.destroy();
+  }
+
+  render() {
+    const {
+      discrete,
+      displayMarkers,
+      value,
+      min,
+      max,
+      step,
+      disabled,
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      className,
+      isRtl,
+      onValueChange,
+      onValueInput,
+      /* eslint-enable @typescript-eslint/no-unused-vars */
+      ...otherProps
+    } = this.props;
+
+    const {
+      markerValue,
+      markerStyles,
+      trackStyleProperty,
+      thumbContainerStyleProperty,
+    } = this.state;
+
+    return (
+      <div
+        className={this.classes}
+        role='slider'
+        aria-valuemin={min}
+        aria-valuemax={max}
+        aria-valuenow={value}
+        data-step={step}
+        aria-label='Select Value'
```

Comment:
we should allow for this to be overridable:

```
aria-label={this.props.ariaLabel}
```

set as a defaultProp to 'Select Value'